src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -147,25 +149,27 @@ class HttpGoogleServicesDAO(priceListUrl: String, defaultPriceList: GooglePriceL
     * @return path to the uploaded GCS object
     */
   override def writeObjectAsRawlsSA(bucketName: GcsBucketName, objectKey: GcsObjectName, objectContents: Array[Byte]): GcsPath = {
-    // if other methods in this class use google2/need these implicits, consider moving to the class level
-    // instead of here at the method level
-    implicit val logger: SelfAwareStructuredLogger[IO] = Slf4jLogger.getLogger[IO]
-
-    // create the storage service, using the Rawls SA credentials
-    // the Rawls SA json creds do not contain a project, so also specify the project explicitly
-    val storageResource = GoogleStorageService.resource(FireCloudConfig.Auth.rawlsSAJsonFile, Option.empty[Semaphore[IO]],
-       project = Some(GoogleProject(FireCloudConfig.FireCloud.serviceProject)))
+    // call the upload implementation
+    val dataStream: Stream[IO, Byte] = Stream.emits(objectContents).covary[IO]
+    streamUploadObject(getStorageResource, bucketName, objectKey, dataStream)
+  }
```

Comment:
I broke up the pre-existing `writeObjectAsRawlsSA` method to extract reusable parts, then invoked those reusable parts from the two `writeObjectAsRawlsSA` signatures.